widget/task_preview.lua

```diff
@@ -0,0 +1,120 @@
+--
+-- Provides:
+-- bling::task_preview::visibility
+--      s   (screen)
+--      v   (boolean)
+--      c   (client)
+--
+local wibox = require("wibox")
+local helpers = require(tostring(...):match(".*bling") .. ".helpers")
+local gears = require("gears")
+local beautiful = require("beautiful")
+local dpi = beautiful.xresources.apply_dpi
+local cairo = require("lgi").cairo
+
+local function draw_widget(c, task_preview_box, screen_radius, widget_bg,
+                           widget_border_color, widget_border_width, margin)
+
+    local content = gears.surface(c.content)
+    local cr = cairo.Context(content)
+    local x, y, w, h = cr:clip_extents()
+    local img = cairo.ImageSurface.create(cairo.Format.ARGB32, w - x, h - y)
+    cr = cairo.Context(img)
+    cr:set_source_surface(content, 0, 0)
+    cr.operator = cairo.Operator.SOURCE
+    cr:paint()
+
+    task_preview_box:setup{
+        {
+            {
+                {
+                    {
+                        image = gears.surface.load(c.icon),
+                        resize = true,
+                        forced_height = dpi(20),
+                        forced_width = dpi(20),
+                        widget = wibox.widget.imagebox
+                    },
+                    {
+                        markup = c.name,
+                        align = "center",
+                        widget = wibox.widget.textbox
+                    },
+                    layout = wibox.layout.align.horizontal
+                },
+                {
+                    {
+                        {
+                            image = gears.surface.load(img),
+                            resize = true,
+                            widget = wibox.widget.imagebox
+                        },
+                        valign = "center",
+                        halign = "center",
+                        widget = wibox.container.place
+                    },
+                    top = margin * 0.25,
+                    widget = wibox.container.margin
+                },
+                fill_space = true,
+                layout = wibox.layout.fixed.vertical
+            },
+            margins = margin,
+            widget = wibox.container.margin
+        },
+        bg = widget_bg,
+        border_width = widget_border_width,
+        border_color = widget_border_color,
```

Comment:
It seems like `border_width` and `border_color` are 4.4 names. In 4.3 it only properly showed up after renaming those two to `shape_border_width` and `shape_border_color`. Those names also worked on 4.4.

Reply:
@Nooo37 Really? I never knew this... this is how I have most of the other widgets done in bling also.